frontend/packages/core/src/Network/index.ts

```diff
@@ -81,6 +98,20 @@ const createClient = () => {
   return axiosClient;
 };
 
+const createProxyClient = () => {
```

Comment:
So we're recreating logic here to create this client so I think we can just make it a bit more customizable to reduce the duplication.

Something like:

```
const successInterceptor = (response: AxiosResponse): AxiosResponse => {
...

const successProxyInterceptor = (response: AxiosResponse): AxiosResponse | Promise<never> => {
...
interface CreateClientProps {
  response?: {
    success: (response: AxiosResponse) => AxiosResponse | Promise<never>;
    error: (error: AxiosError) => Promise<ClutchError>;
  };
}

const createClient = ({ response }: CreateClientProps): AxiosInstance => {
  const axiosClient = axios.create({
    // n.b. the client will treat any response code >= 400 as an error and apply the error interceptor.
    validateStatus: status => {
      return status < 400;
    },
  });

  if (response) {
    axiosClient.interceptors.response.use(response.success, response.error);
  }

  return axiosClient;
};

const client = createClient({ response: { success: successInterceptor, error: errorInterceptor } });
const proxyClient = createClient({
  response: { success: successProxyInterceptor, error: errorInterceptor },
});
```

Haven't tested this specifically but it should work and then we're just telling it which interceptors to use versus recreating the instance.

Reply:
Changes done!